unlock-app/src/components/interface/checkout/main/Select.tsx

```diff
@@ -337,7 +341,20 @@ export function Select({ checkoutService }: Props) {
 
   const isLoading = isLocksLoading || isLoadingHook || isMembershipsLoading
 
+  const { connected } = useAuth()
+  const { signIn, isSignedIn } = useSIWE()
+  const useDelegatedProvider = paywallConfig?.useDelegatedProvider
+
   useEffect(() => {
+    const signToSignIn = async () => {
+      await signIn()
+    }
+
+    if (!connected && useDelegatedProvider) {
+      console.log('Not connected')
```

Comment:
```suggestion
```

Reply:
I wonder if we should move this logic "higher" in the hierarchy because I worry that when we refactor the `Select` step we will miss it. Can you move it higher please?